Feature/integrator

idovgalyov-4xxi opened the pull request:
Added module Integrator and supplements for it.
Provided Readme-s with description how to launch and what happens after that
This module also can be modified in future by requirements of the main module (yet to be done).

idovgalyov-4xxi committed:
Workin' progress

Still.
Although the main part is complete.

smirik reviewed (changes requested):
The file asteroids_format.csv should be fully removed from the repository (not like git delete, but fully, with rewriting history).

The reason is that git is a bad place to keep large files. It makes the repository extremely large. Plus, the file can be changed in future, so we'll need to update ~100 Mb when apply the changes.

Please, ask @pluseg for help with git.

smirik commented on integrator/integrator.py:
```diff
@@ -0,0 +1,98 @@
+from integratorlib import *
+
+
+# Pack size for integrator
```
All code should be moved into the functions. There is too much code out of the function that makes the testing very difficult. 

Use the following rules (they are not mandatory, however, they are useful):

1. Each function should have <= 15 lines of the code.
2. Each file should have <=200 lines of the code.
3. The code out of the functions (except imports and definitions) should have <=15 lines of the code.

smirik commented on integrator/integrator.py:
```diff
@@ -0,0 +1,98 @@
+        try:
+            astlist=[str(i) for i in range(int(argv[2]),int(argv[3])+1)]
+        except:
+            quit('Error! Wrong range arguments. Please follow the format described in documentation')
```
quit is bad option for python software. Throw & process an exception instead of that & return the proper results in a function.

smirik commented on integrator/integrator.py:
```diff
@@ -0,0 +1,98 @@
+    ')---------------------------------------------------------------------\n',
+    ' style (Cartesian, Asteroidal, Cometary) = Asteroidal\n',
+    ')--------------------------------------------d or D is not matter--0d0 is possible too--\n',' \n']
+if len(smallin)<6:
```
how is it possible? You define smallin array a line before. Why do you check the number of lines. It can't be less than 6.

smirik replied:
You can check this in a test if you like to.

smirik commented on integrator/integrator.py:
```diff
@@ -0,0 +1,98 @@
+    proc(smallin,sheader)
+    k=0
+
+quit('.aei files for the list created.')
```
The file should not end with quit.

smirik commented on integrator/integratorlib.py:
```diff
@@ -0,0 +1,65 @@
+from subprocess import Popen
+
+
+def find_asteroid(i,f2,lrec,stlen):
```
Please, use the good names of the variables:

- i -> asteroid_name
- f2 -> source_file
- lrec -> num_lines
- stlen -> ...

smirik commented on integrator/integratorlib.py:
```diff
@@ -0,0 +1,65 @@
+    print('\n\tSeeking record about ',repr(i))
+    st=[]
+    flag=0
+    a=0
```
Please, use the normal names of the variables. a is not clean.

smirik commented on integrator/integratorlib.py:
```diff
@@ -0,0 +1,65 @@
+    return st
+
+
+def proc(smallin,sheader):
```
Please, consider the renaming of the function. I don't understand what "proc" function should do.

smirik replied:
Also, smallin -> what is that? Array? File? A better name should be used.

smirik commented on integrator/integratorlib.py:
```diff
@@ -0,0 +1,65 @@
+    p=Popen("cd ../mercury; mv *.aei ../aeibase/ ; ./simple_clean.sh",shell=True)
+    p.wait()
+    print('\t.aei files created.')
+    return None
```
The function should return something usually. It is a bad idea to return nothing.

smirik reviewed (changes requested):
The changes should be applied before further review. The code must be reviewed again.

idovgalyov-4xxi committed:
Module integrator

idovgalyov-4xxi committed:
Integrator-Classifier-Compositor (beta)

Here are working versions of modules: integrator.f90, libration.f90 and compositor.f90.
Some job still remains.
Unit tests, ReadMe's, comments and estetical changes will be provided soon later.

idovgalyov-4xxi committed:
Local update

Workin' progress

All needed functions are stable, but need some modifications
(e. g. linked list support for libration and resonance finder modules).

For now, only intergator is enabled for solving priority tasks.

idovgalyov-4xxi committed:
Global update

Global restructurization of subroutines and functions.
Provided three-body case for processing.
Provided keys in global parameters for setting the program.
Libration classifier is more smart and can ignore the most part of slow circulations (but still needs calibration, I think).
This version is stable and almost ready for a work.
Unit tests will be provided with next commit.

idovgalyov-4xxi committed:
Unit tests, descriptions and small changes

Provided unit tests and description in Readme.md
Some small corrections

idovgalyov-4xxi committed:
bugfix

Fixed a mistake in compositor.f90 where one if-clause had incorrect syntax

idovgalyov-4xxi committed:
Bugfix and other

Fixed an error when compositor tried to find previously unfound asteroid several times.
Function norm_ang became pure for supporting "forall" construction in "resph" subroutine in module Librations.
Excluded Pluto from calculations (however, left calculations of id_matrices).

smirik committed:
Add blank after stop. Add comp.x to ignore.

idovgalyov-4xxi committed:
Big update (several features)

*) Corrected subroutine "count_axis_3body" now using Jupiter parameters in formulae.
*) Parameter "pwd" now determines automatically
*) Reduced parameter "kmax" in module Integrator to a value 100.
*) Created directory for planet .aei files separatedly from the others.
*) Added some scripts for plotting if needed (if using, it makes getting graphics more comfortable).

idovgalyov-4xxi committed:
Big update

The libration classifier was improved, using more precise cross-periodogram checking and libration detecting parameters.
Some procedures have been separated and placed in a new module "librations_suport.f90" (corrected unit tests will be provided in the next commit).

idovgalyov-4xxi committed:
Small update

Added common output file.
Updated unit tests and ReadMe.

idovgalyov-4xxi committed:
Version for server

Added script that produces parallel calculations and gathers results in a common directories.

idovgalyov-4xxi committed:
Small bugfix

Fixed a bug when the Main script stopped working after first 1000 asteroids.
Also changed implicit mode by swithing "use only integration" because the classifier needs to be improved.

idovgalyov-4xxi committed:
Technical update

Found that logical core existance in processor should be taken into account.
Improved the classifier: now cross-periodigram is not checked on periods close to orbital periods of objects.
Fixed the bug in function "count_axis_3body" and prepared a pattern for the next update (still need to get second expression for precession rate).

idovgalyov-4xxi committed:
Update integrator.f90

idovgalyov-4xxi committed:
Small script update

Optimized block sepatation according to a number of available CPU cores and max size of an integration block.

idovgalyov-4xxi committed:
Merge branch 'feature/integrator' of https://github.com/4xxi/resonances-fortran into feature/integrator

global repo was updated

idovgalyov-4xxi committed:
2017-05-15 Small update

Fixed the case when some boundary .aei files could have been computed twice.
Made some optimization of separating by blocks.

idovgalyov-4xxi committed:
Global update

Updated resonant axis calculator with more correct expressions for precession rate.
Included calculation of proper semiaxis and accounting problems while finding it.
Corrected some calculating mistakes in the classifier.

idovgalyov-4xxi committed:
The great update

The most important changes:
1) Potential resonance list for asteroid is not saved in scratch files anymore
2) The procedure of calculating resonance strength (Gallardo, 2014) has been included
3) The resonance classifier has been improved. The correlation is checked in local sections separately

Temporary disabled metadata and graphics.
